SyslogAppenderCustomLayoutTest is unreliable

## Description

SyslogAppenderCustomLayoutTest is unreliable

```
[ERROR] Failures:
[ERROR]   SyslogAppenderCustomLayoutTest>SyslogAppenderTest.testTCPAppender:56->SyslogAppenderTestBase.sendAndCheckLegacyBsdMessage:74->SyslogAppenderTestBase.checkTheNumberOfSentAndReceivedMessages:112 The number of received messages should be equal with the number of sent messages expected:<1> but was:<0>
[ERROR]   SyslogAppenderCustomLayoutTest>SyslogAppenderTest.testUDPAppender:79->SyslogAppenderTestBase.sendAndCheckLegacyBsdMessage:74->SyslogAppenderTestBase.checkTheNumberOfSentAndReceivedMessages:112 The number of received messages should be equal with the number of sent messages expected:<1> but was:<0>
```

## Configuration

**Version:** 2.x at 01125b11a7c52af12204a1c4a646158f63b2d6eb

**Operating system:** 
Microsoft Windows [Version 10.0.19045.4780

**JDK:** 
Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)
Maven home: C:\java\apache-maven-3.9.9
Java version: 17.0.12, vendor: Eclipse Adoptium, runtime: C:\Program Files\Eclipse Adoptium\jdk-17.0.12.7-hotspot
Default locale: en_US, platform encoding: UTF-8
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

## Logs

```
[ERROR] Tests run: 5, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 3.549 s <<< FAILURE! -- in org.apache.logging.log4j.core.appender.SyslogAppenderCustomLayoutTest
[INFO] Running org.apache.logging.log4j.core.appender.SmtpAppenderAsyncTest
10:07:11.688 main ERROR The pattern, literal, and isEventTimestamp attributes are mutually exclusive.
[ERROR] org.apache.logging.log4j.core.appender.SyslogAppenderCustomLayoutTest.testTCPAppender -- Time elapsed: 0.225 s <<< FAILURE!
java.lang.AssertionError: The number of received messages should be equal with the number of sent messages expected:<1> but was:<0>
        at org.apache.logging.log4j.core.appender.SyslogAppenderTestBase.checkTheNumberOfSentAndReceivedMessages(SyslogAppenderTestBase.java:112)
        at org.apache.logging.log4j.core.appender.SyslogAppenderTestBase.sendAndCheckLegacyBsdMessage(SyslogAppenderTestBase.java:74)
        at org.apache.logging.log4j.core.appender.SyslogAppenderTest.testTCPAppender(SyslogAppenderTest.java:56)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

[ERROR] org.apache.logging.log4j.core.appender.SyslogAppenderCustomLayoutTest.testUDPAppender -- Time elapsed: 0.209 s <<< FAILURE!
java.lang.AssertionError: The number of received messages should be equal with the number of sent messages expected:<1> but was:<0>
        at org.apache.logging.log4j.core.appender.SyslogAppenderTestBase.checkTheNumberOfSentAndReceivedMessages(SyslogAppenderTestBase.java:112)
        at org.apache.logging.log4j.core.appender.SyslogAppenderTestBase.sendAndCheckLegacyBsdMessage(SyslogAppenderTestBase.java:74)
        at org.apache.logging.log4j.core.appender.SyslogAppenderTest.testUDPAppender(SyslogAppenderTest.java:79)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

## Reproduction

```
mvn clean install
```

--- a/log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/SyslogAppenderTestBase.java
+++ b/log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/SyslogAppenderTestBase.java
@@ -16,6 +16,8 @@
  */
 package org.apache.logging.log4j.core.appender;
 
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.awaitility.Awaitility.await;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
@@ -49,7 +51,7 @@ abstract class SyslogAppenderTestBase {
             "TestApp - Audit [Transfer@18060 Amount=\"200.00\" FromAccount=\"123457\" ToAccount=\"123456\"]"
                     + "[RequestContext@18060 ipAddress=\"192.168.0.120\" loginId=\"JohnDoe\"] Transfer Complete";
     protected LoggerContext ctx = LoggerContext.getContext();
-    protected static final int DEFAULT_TIMEOUT_IN_MS = 100;
+    protected static final int DEFAULT_TIMEOUT_IN_MS = 3000;
     protected MockSyslogServer syslogServer;
     protected SyslogAppender appender;
     protected Logger root = ctx.getLogger("SyslogAppenderTest");
@@ -61,7 +63,7 @@ abstract class SyslogAppenderTestBase {
         LoggerContext.getContext().reconfigure();
     }
 
-    protected void sendAndCheckLegacyBsdMessages(final List<String> messagesToSend) throws InterruptedException {
+    protected void sendAndCheckLegacyBsdMessages(final List<String> messagesToSend) {
         for (final String message : messagesToSend) {
             sendDebugLegacyBsdMessage(message);
         }
@@ -69,7 +71,7 @@ abstract class SyslogAppenderTestBase {
         checkTheEqualityOfSentAndReceivedMessages(Level.DEBUG);
     }
 
-    protected void sendAndCheckLegacyBsdMessage(final String message) throws InterruptedException {
+    protected void sendAndCheckLegacyBsdMessage(final String message) {
         sendDebugLegacyBsdMessage(message);
         checkTheNumberOfSentAndReceivedMessages();
         checkTheEqualityOfSentAndReceivedMessages(Level.DEBUG);
@@ -80,7 +82,7 @@ abstract class SyslogAppenderTestBase {
         root.debug(message);
     }
 
-    protected void sendAndCheckStructuredMessages(final int numberOfMessages) throws InterruptedException {
+    protected void sendAndCheckStructuredMessages(final int numberOfMessages) {
         for (int i = 0; i < numberOfMessages; i++) {
             sendInfoStructuredMessage();
         }
@@ -88,7 +90,7 @@ abstract class SyslogAppenderTestBase {
         checkTheEqualityOfSentAndReceivedMessages(Level.INFO);
     }
 
-    protected void sendAndCheckStructuredMessage() throws InterruptedException {
+    protected void sendAndCheckStructuredMessage() {
         sendInfoStructuredMessage();
         checkTheNumberOfSentAndReceivedMessages();
         checkTheEqualityOfSentAndReceivedMessages(Level.INFO);
@@ -108,15 +110,13 @@ abstract class SyslogAppenderTestBase {
         root.info(MarkerManager.getMarker("EVENT"), msg);
     }
 
-    protected void checkTheNumberOfSentAndReceivedMessages() throws InterruptedException {
-        assertEquals(
-                sentMessages.size(),
-                getReceivedMessages(DEFAULT_TIMEOUT_IN_MS).size(),
-                "The number of received messages should be equal with the number of sent messages");
+    protected void checkTheNumberOfSentAndReceivedMessages() {
+        await().atMost(DEFAULT_TIMEOUT_IN_MS, MILLISECONDS)
+                .until(() -> syslogServer.getMessageList().size() == sentMessages.size());
     }
 
-    protected void checkTheEqualityOfSentAndReceivedMessages(final Level expectedLevel) throws InterruptedException {
-        final List<String> receivedMessages = getReceivedMessages(DEFAULT_TIMEOUT_IN_MS);
+    protected void checkTheEqualityOfSentAndReceivedMessages(final Level expectedLevel) {
+        final List<String> receivedMessages = getReceivedMessages();
 
         assertNotNull(receivedMessages, "No messages received");
         for (int i = 0; i < receivedMessages.size(); i++) {
@@ -148,10 +148,7 @@ abstract class SyslogAppenderTestBase {
         root.setLevel(Level.DEBUG);
     }
 
-    protected List<String> getReceivedMessages(final int timeOutInMs) throws InterruptedException {
-        synchronized (syslogServer) {
-            syslogServer.wait(timeOutInMs);
-        }
+    protected List<String> getReceivedMessages() {
         return syslogServer.getMessageList();
     }
 

